src/Features/Core/Portable/Snippets/AbstractSnippetService.cs

```diff
@@ -46,10 +49,23 @@ public ISnippetProvider GetSnippetProvider(string snippetIdentifier)
         /// </summary>
         public async Task<ImmutableArray<SnippetData>> GetSnippetsAsync(Document document, int position, CancellationToken cancellationToken)
         {
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
surprised you don't need GetRequiredSyntaxTree here.

Reply:
`IsInNonUserCode` handles null:

https://github.com/dotnet/roslyn/blob/447c48956e16f4286db0a457840ab9f5dae1a17c/src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSyntaxFactsService.cs#L28-L36

It's not annotated though.

Reply:
The code is an exact copy of `GetSnippetDataAsync` anyway. Just pulled the code from `GetSnippetDataAsync` which is called for each provider to a path that's called once.